src/resources/views/base/inc/alerts.blade.php

```diff
@@ -1,21 +1,43 @@
 {{-- Bootstrap Notifications using Prologue Alerts & PNotify JS --}}
 <script type="text/javascript">
-  Noty.overrideDefaults({
-    layout   : 'topRight',
-    theme    : 'backstrap',
-    timeout  : 2500, 
-    closeWith: ['click', 'button'],
-  });
 
-  @foreach (\Alert::getMessages() as $type => $messages)
+// This is intentionaly run after dom loads so this way we can avoid showing duplicate alerts
+// when the user is beeing redirected by persistent table, that happens before this event triggers.
 
-      @foreach ($messages as $message)
+window.addEventListener('DOMContentLoaded', function() {
+    //get the php alerts
+    var $crudAlerts = JSON.parse('{!! json_encode( \Alert::getMessages() , JSON_HEX_APOS) !!}');
 
-        new Noty({
-          type: "{{ $type }}",
-          text: "{!! str_replace('"', "'", $message) !!}"
-        }).show();
+        Noty.overrideDefaults({
+            layout   : 'topRight',
+            theme    : 'backstrap',
+            timeout  : 2500,
+            closeWith: ['click', 'button'],
+        });
 
-      @endforeach
-  @endforeach
-</script>
\ No newline at end of file
+
+    for (var type in $crudAlerts) {
+        for(var message in $crudAlerts[type]) {
+            new Noty({
+                    type: type,
+                    text: $crudAlerts[type][message]
+                }).show();
+        }
+    }
+    //if the user was redirected by persistent table we get any alerts to show from localstorage
+    //as they are not in session anymore but we previously stored them.
+    $passedAlerts = JSON.parse(localStorage.getItem('passAlerts'));
+
+    if(typeof $passedAlerts === 'object' && $passedAlerts !== null) {
+        $passedAlerts.forEach(function($item) {
+                new Noty({
+                    type: $item.type,
+                    text: $item.text
+                }).show();
+        });
+    }
```

Comment:
Hmmm feels like there's some code duplication here. I think we can avoid calling `new Noty` twice, by adding the results of `localStorage.getItem('passAlerts')` to `backpackAlerts`. So right after we do `var $crudAlerts = JSON.parse('{!! json_encode( \Alert::getMessages() , JSON_HEX_APOS) !!}');` we could just manipulate `$crudAlerts` further, by adding to that array whatever's stored inside localstorage.